src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -234,6 +234,11 @@ class LicenceService(
 
   fun activateLicences(licenceIds: List<Long>) {
     val matchingLicences = licenceRepository.findAllById(licenceIds)
+    matchingLicences.forEach { licence ->
+      if (licence.statusCode != APPROVED) {
+        throw ValidationException("Licence with ID ${licence.id} cannot be activated because it is not currently in the APPROVED state")
+      }
+    }
```

Comment:
Fail the whole batch when 1 is not APPROVED? I know was my suggestion to check state, but it might be a bit drastic to fail all if one is not APPROVED?

Reply:
What would you suggest? Just log it and carry on?

Reply:
Could we skip any that are not now approved? Or just ignore this, as its unlikely to happen between selecting the list and applying the activate.

Reply:
Pushed change, now skip any that are not approved but not fail